### DISCUSSION TITLE
How to safely download a PDF

### DISCUSSION THREAD

**Discussion by librewolffy** [netblue30/firejail#5180] [Q&A] [answered]
Operating system is Linux Mint 20.2. 
Latest version of Librewolf and Firejail.

I am using Firejail to get Librewolf. 

How do I safely download a PDF file?

Does Firejail cover everything from downloading PDFs?

**Comment by rusty-snake** [comment, marked as answer]
What do you mean with "safely download"? Downloading isn't danger (beside network access).

> Does Firejail cover everything from downloading PDFs and even watching Youtube?

Cover what? What's your thread model?

**Reply by librewolffy** [reply]
"safely download" means there would be no malware or viruses attached to the PDF files:

**Reply by rusty-snake** [reply]
> "safely download" means there would be no malware or viruses attached to the PDF files:

firejail = sandbox = if there is malware it has limited scope (permissions) where it can harm.
virus-scanner = notifies (and blocks) files containing malware.

The AV (virus-scanner) security model works by blocking files containing known malware signatures. This works good for *known* malware but does not block downloads containing unknown (=new or obfuscated) malware.

The sandbox security model works by restricting the permissions of the pdf/mp4/... viewer/player so that if a file contains malware that exploits a bug in your pdf-viewer/video-player/web-browser/... it has no access to your sensitive data, can not write to e.g. bashrc to become persistent, can not download additional payloads from the internet, can not create W&X memory mappings, ... sandbox configuration ... .

> Would it be better to just watch from the website instead of downloading the video to my system and watch?

Security: yes (if you open the website anyway for downloading)
Privacy: no

**Comment by rusty-snake** [comment]
:upside_down_face: :upside_down_face: :upside_down_face: :upside_down_face: :upside_down_face: :upside_down_face: :upside_down_face: :upside_down_face: :upside_down_face: 

https://www.privateproxyguide.com/best-antivirus-for-linux/:

> 3.2 [Firetools](https://firejailtools.wordpress.com/)
>
> [... snip ...]
> 
> Basic Features
>
> - Reliable sandboxing for web browsers.
> - ***Advanced real-time virus and malware scanning.***

When did we add the advanced version? I did not even know that we have a basic version. LOL


:upside_down_face: :upside_down_face: :upside_down_face: :upside_down_face: :upside_down_face: :upside_down_face: :upside_down_face: :upside_down_face: :upside_down_face:
